client/console/src/main/java/org/apache/syncope/client/console/audit/AuditHistoryDirectoryPanel.java

```diff
@@ -0,0 +1,187 @@
+import org.apache.syncope.client.console.wicket.extensions.markup.html.repeater.data.table.KeyPropertyColumn;
+import org.apache.syncope.client.console.wicket.markup.html.bootstrap.dialog.BaseModal;
+import org.apache.syncope.client.console.wicket.markup.html.form.ActionLink;
+import org.apache.syncope.common.lib.to.AnyTO;
+import org.apache.syncope.common.lib.to.AuditEntryTO;
+import org.apache.syncope.common.lib.to.GroupTO;
+import org.apache.syncope.common.lib.to.UserTO;
+import org.apache.syncope.common.lib.types.AnyTypeKind;
+import org.apache.syncope.common.lib.types.AuditElements;
+import org.apache.wicket.PageReference;
+import org.apache.wicket.extensions.markup.html.repeater.data.table.IColumn;
+import org.apache.wicket.extensions.markup.html.repeater.data.table.PropertyColumn;
+import org.apache.wicket.extensions.markup.html.repeater.util.SortParam;
+import org.apache.wicket.model.IModel;
+import org.apache.wicket.model.StringResourceModel;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class AuditHistoryDirectoryPanel extends
+    DirectoryPanel<AuditEntryBean, AuditEntryBean, DirectoryDataProvider<AuditEntryBean>, AuditHistoryRestClient>
+    implements ModalPanel {
+
+    private static final long serialVersionUID = -8248734710505211261L;
+
+    private final BaseModal<?> baseModal;
+
+    private final MultilevelPanel multiLevelPanelRef;
+
+    private final AnyTO anyTO;
+
+    private final AnyTypeKind anyTypeKind;
+
+    public AuditHistoryDirectoryPanel(
+        final BaseModal<?> baseModal,
+        final MultilevelPanel multiLevelPanelRef,
+        final PageReference pageRef,
+        final AnyTO anyTO) {
+
+        super(MultilevelPanel.FIRST_LEVEL_ID, pageRef);
+        disableCheckBoxes();
+        
+        this.baseModal = baseModal;
+        this.multiLevelPanelRef = multiLevelPanelRef;
+        this.anyTO = anyTO;
+
+        if (anyTO instanceof UserTO) {
+            anyTypeKind = AnyTypeKind.USER;
+        } else if (anyTO instanceof GroupTO) {
+            anyTypeKind = AnyTypeKind.GROUP;
+        } else {
+            anyTypeKind = AnyTypeKind.ANY_OBJECT;
+        }
+        this.restClient = new AuditHistoryRestClient();
+        initResultTable();
+    }
+
+    @Override
+    protected DirectoryDataProvider<AuditEntryBean> dataProvider() {
+        return new AuditHistoryProvider(rows);
+    }
+
+    @Override
+    protected String paginatorRowsKey() {
+        return Constants.PREF_AUDIT_HISTORY_PAGINATOR_ROWS;
+    }
+
+    @Override
+    protected List<IColumn<AuditEntryBean, String>> getColumns() {
+        final List<IColumn<AuditEntryBean, String>> columns = new ArrayList<>();
+        columns.add(new KeyPropertyColumn<>(
+            new StringResourceModel("key", this), "key"));
```

Comment:
There is no need of a `key` column: all rows will always have the same value, matching the user / group / any object selected.